include/spdlog/details/os.h

```diff
@@ -128,6 +129,65 @@ constexpr inline unsigned short eol_size()
 }
 #endif
 
+//Return true and list if filename contains relative
+inline bool dir_check(const std::string& filename, std::list<std::string>& dirs)
+{
+#ifdef __linux__
+    std::string directory;
+    std::string::size_type index, previndex = 0, size;
+
+    index = filename.find("/", previndex);
+    if(!index)
+        dirs.push_back("/");
+
+    do {
+        if(index - previndex == 0) 
+            directory = filename.substr(previndex, index - previndex + 1);
+        else 
+            directory = filename.substr(previndex, index - previndex);
+
+        if(directory != "/")
+            dirs.push_back(directory);
+
+        previndex = index + 1;
+        index = filename.find("/", previndex);
+    }while(index != std::string::npos);
+
+    return true;
+
+#endif
+
+}
+
+//Create directories by referring to list 
+//Return 0 if success 
+inline bool create_dirs(std::list<std::string>& dirs)
+{
+#ifdef __linux__
```

Comment:
Should be on windows too as spdlog supports windows and should behave the same where possible on all supported plaforms